### DISCUSSION TITLE
ui.leaflet in tab panel, TileLayer error on clear

### DISCUSSION THREAD

**Discussion by thickmn** [zauberzeug/nicegui#2587] [Q&A]
### Question

I have a tab with a leaflet map in a dialog that puts some points within a radius. When a user clicks outside of the dialog or closes the dialog, I'd usually clear the tab panel. Now that the leaflet map is inside of the dialog, I get:

TypeError: unhashable type: 'TileLayer'

Any advice on properly clearing a tab panel? 

```py
from nicegui import ui

class Content:
    def build(self):
        self._tab_panels = ui.tab_panels(animated=False)

        with self._tab_panels:
            self.create_map_tab()

    def create_map_tab(self):
        map_instance = ui.leaflet((-51, 82), zoom=9)
        map_instance.generic_layer(name='circle', args=[(-51, 82), {'color': 'green', 'radius': 1000}])

    def clear_tab_panels(self):
        self._tab_panels.clear()

def simulate_error():
    content = Content()
    content.build()
    content.clear_tab_panels()

simulate_error()

ui.run()
```

**Comment by falkoschindler** [comment]
Hi @thickmn,

I think this problem can be boiled down to
```py
with ui.card() as card:
    m = ui.leaflet((-51, 82))
    m.marker(latlng=(-51, 82))
card.clear()
```

On the main branch the TypeError is already fixed by PR https://github.com/zauberzeug/nicegui/pull/2544.
But there's still a KeyError occurring. Maybe it's related to PR https://github.com/zauberzeug/nicegui/pull/2603. I'll look into it.

**Comment by falkoschindler** [comment]
There was another issue (see #2609), which will be fixed in the upcoming version 1.4.16.

**Comment by thickmn** [comment]
thanks @falkoschindler !

**Comment by falkoschindler** [comment]
Version [1.4.16](https://github.com/zauberzeug/nicegui/releases/tag/v1.4.16) has just been released. 🎉
